language/move-analyzer/src/completion.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::context::Context;
+use lsp_server::Request;
+use lsp_types::{CompletionItem, CompletionItemKind, CompletionParams, Position};
+use move_command_line_common::files::FileHash;
+use move_lang::{
+    diagnostics::Diagnostic,
+    parser::lexer::{Lexer, Tok},
+};
+use std::collections::HashSet;
+
+/// A static list of all possible keywords in the Move language, including contextual ones.
+const KEYWORDS: &[&str] = &[
```

Comment:
The documentation generator also keeps a [list of keywords](https://github.com/diem/diem/blob/main/language/move-prover/docgen/src/docgen.rs#L34). It might be good to use those keywords, or see if this list could be centralized in one place.

Reply:
I agree! I also like how docgen categorizes them as keywords and "weak keywords."

But hmm, what's a central location we can place these lists... I think it makes sense to place the lists within `language/move-lang` somewhere, since to me that's the crate that defines everything related to the Move source language. But maybe some people would think it strange that the crate wouldn't use the lists of keywords itself.

For lack of a better idea, I'll update this PR to define the lists as `move_lang::parser::keywords::KEYWORDS` and `move_lang::parser::keywords::CONTEXTUAL_KEYWORDS` (is "contextual" a better name than "weak"? I personally think so).

Reply:
Having these defined in the Move land parser/lexer makes sense to me. Also partial to `contextual` rather than `weak`. Although  this then makes me wonder if each of these keywords should be tagged with their context in some way (e.g., `spec`). But that can wait till a different PR/when context-specific completions are added :)